types/webpack-dev-server/index.d.ts

```diff
@@ -353,7 +356,11 @@ declare module 'webpack' {
     }
 }
 
+type Socket = sockjs.Connection|WebSocket;
```

Comment:
I"m not sure about wiring types to specific implementation. In the package that is left to be resolved by dynamic require. It also allows user implementation:
https://github.com/webpack/webpack-dev-server/blob/b5b9cb47fe6d930e97a96fc49d80d3df9d3cae4a/lib/utils/getSocketServerImplementation.js
All those impl creates connections, that extend EventEmitter in the chain I think.
What do you think?

Reply:
You're absolutely right - I've updated to use the `NodeJS.EventEmitter` interface 🙇